### PR TITLE
Fix incorrect miner response in some cases

### PR DESCRIPTION
**Pull request by MsRandom**
Fix miner preserving some state between requests, along with adding versioning to validators

### CODE REVIEW

File: validator/src/validator/mod.rs [hunk 1]
```diff
@@ -31,9 +31,10 @@ mod memory_storage;
 mod event_future;
 mod neuron_data;
 
-const VERSION_KEY: u64 = 1;
+const WEIGHTS_VERSION: u64 = 1;
```
Comment (3xela) [inline validator/src/validator/mod.rs:34]: what is this

File: validator/src/validator/mod.rs [hunk 2]
```diff
@@ -531,7 +550,7 @@ impl Validator {
                         }
                     }
 
-                    (&mut current_row[range.start..range.start + len]).copy_from_slice(&buffer[..len]);
+                    (&mut current_row[read_chunk_range]).copy_from_slice(&buffer[..len]);
```
Comment (3xela) [inline validator/src/validator/mod.rs:553]: why did you change this

**Review by 3xela — COMMENTED** [review]
just a few questions

File: validator/src/validator/mod.rs (hunk 2 above)
Comment (MsRandom) [reply validator/src/validator/mod.rs:553]: Just seemed cleaner

File: validator/src/validator/mod.rs (hunk 1 above)
Comment (MsRandom) [reply validator/src/validator/mod.rs:34]: It's to be updated when we want to update the `weights_version` hyperparameter on chain to force validators to be on a newer version of the codebase

File: validator/src/validator/mod.rs (hunk 2 above)
Comment (3xela) [reply validator/src/validator/mod.rs:553]: makes sense

File: validator/src/validator/mod.rs (hunk 1 above)
Comment (3xela) [reply validator/src/validator/mod.rs:34]: i see

**Review by 3xela — APPROVED** [review]
looks good
